### PR TITLE
PIN-4473 BKE - BFF Added new file upload manager

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala [hunk 1]
```diff
@@ -134,4 +134,16 @@ object BFFErrors {
 
   final case class InstitutionNotFound(selfcareId: UUID)
       extends ComponentError("0033", s"Institution $selfcareId not found")
+
+  final case class UploadDocumentClientError(eServiceId: UUID, descriptorId: UUID)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala]:
```suggestion
  final case class CreateDocumentBadRequest(eServiceId: UUID, descriptorId: UUID)
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala [hunk 2]
```diff
@@ -134,4 +134,16 @@ object BFFErrors {
...
+      extends ComponentError(
+        "0034",
+        s"Error < 500 Uploading document to Eservice $eServiceId and descriptor $descriptorId"
+      )
+
+  final case class UploadDocumentServerError(eServiceId: UUID, descriptorId: UUID, filePath: String)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala]:
```suggestion
  final case class CreateDocumentUnexpectedError(eServiceId: UUID, descriptorId: UUID, filePath: String)
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala [hunk 3]
```diff
@@ -134,4 +134,16 @@ object BFFErrors {
 
   final case class InstitutionNotFound(selfcareId: UUID)
       extends ComponentError("0033", s"Institution $selfcareId not found")
+
+  final case class UploadDocumentClientError(eServiceId: UUID, descriptorId: UUID)
+      extends ComponentError(
+        "0034",
+        s"Error < 500 Uploading document to Eservice $eServiceId and descriptor $descriptorId"
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala]:
```suggestion
        s"Bad request creating document metadata for Eservice $eServiceId and descriptor $descriptorId"
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala [hunk 4]
```diff
@@ -134,4 +134,16 @@ object BFFErrors {
...
+      )
+
+  final case class UploadDocumentServerError(eServiceId: UUID, descriptorId: UUID, filePath: String)
+      extends ComponentError(
+        "0035",
+        s"Error >= 500 Uploading document with path $filePath to Eservice $eServiceId and descriptor $descriptorId"
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala]:
```suggestion
        s"Unexpected error creating document metadata for Eservice $eServiceId and descriptor $descriptorId with path $filePath"
```

**Review by galales — COMMENTED** [review]
looks good, minor changes suggested

**Review by beetlecrunch — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**